### PR TITLE
Add missing crate docs and fix doc url syntax

### PR DESCRIPTION
**Pull request by jtescher**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ adapter crates to assist in propagating state and instrumenting applications.
 
 In particular, the following crates are likely to be of interest:
 
+- [`opentelemetry-http`] provides an interface for injecting and extracting
```
Comment (frigus02) [inline README.md:66]: `opentelemetry-http` and `http` don't have corresponding links

```markdown
[`opentelemetry-http`]: https://crates.io/crates/opentelemetry-http
[`http`]: https://crates.io/crates/http
```

File: README.md [hunk 2]
```diff
@@ -71,8 +73,10 @@ In particular, the following crates are likely to be of interest:
   metrics information to [`Prometheus`].
 - [`opentelemetry-zipkin`] provides a pipeline and exporter for sending trace
   information to [`Zipkin`].
-- [`opentelemetry-contrib`] provides additional exporters to vendors like
-  [`Datadog`].
+- [`opentelemetry-datadog`] provides additional exporters to [`Datadog`].
```
Comment (frigus02) [inline README.md:76]: `opentelemetry-datadog` doesn't have a corresponding link

```markdown
[`opentelemetry-datadog`]: https://crates.io/crates/opentelemetry-datadog
```

File: README.md (hunk 1 above)
Comment (frigus02) [reply README.md:66]: Still missing the a corresponding link for `http` 🙂 

```markdown
[`http`]: https://crates.io/crates/http
```

Comment (jtescher) [reply README.md:66]: ok should have 'em all now 😅

Comment (frigus02) [reply README.md:66]: Looks good 😄

**Review by frigus02 — APPROVED** [review]

**Pull request merged**